test/test_querying.py

```diff
@@ -152,3 +150,15 @@ def test_get_frequencies(session):
     df = pd.DataFrame.from_dict({"frequency": [None, "1 monthly", "1 yearly"]})
 
     assert_frame_equal(r, df)
+
+def test_disambiguation_by_frequency(session):
+
+    with pytest.warns(UserWarning):
```

Comment:
I think there's a form of this that lets you match on part of the warning text, just to make sure we're picking up the frequency warning and not the file one?

Reply:
Good point